kbi/0028/index.rst

```diff
@@ -0,0 +1,209 @@
+
+Named user share
+^^^^^^^^^^^^^^^^
+
+If a folder is shared with a named user, they will see it in their own
+account like any other folder. In principle, access for a share
+recipient would be analogous to that of an owner, and use an URL
+starting with:
+
+.. code-block:: none
+
+   https://example.com/nextcloud/remote.php/dav/files/USERNAME/
+
+However, with Nextcloud (Sciebo) being a federated service, each user
+may have a different instance URL to access their data. Additionally,
+the URL includes the username, and each user may place the shared
+directory in a different place within their home directory.
+
+Public share, password protected
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For a folder shared with a password-protected link, the access URLs
+would start with:
+
+.. code-block:: none
+
+   https://example.com/nextcloud/public.php/webdav
+
+The share token (part of the share link) needs to be provided as
+username, and the (optional) share password as password. Note that
+these are sent as request credentials, and are not included in the
+URL.
+
+URL pattern - summary
+^^^^^^^^^^^^^^^^^^^^^
+
+In summary, it is useful to represent the WebDAV URL as a combination
+of the following components:
+
+.. code-block:: none
+
+   <instance>/<accesspath>/<dirpath>/<filepath>
+
+where:
+
+* ``<instance>`` is the instance URL
+  (``https://example.com/nextcloud/`` in given examples)
+* ``<accesspath>`` is either ``remote.php/dav/files/USERNAME/`` or
+  ``public.php/webdav``
+* ``<dirpath>`` is the path to the shared folder in user's home
+  directory (none for public shares)
+* ``<filepath>`` is the path to a particular file relative to the
+  shared folder (``<dirpath>``)
+
+Listing files
+-------------
+
+For generating the dataset, a list of file names (relative paths) and
+their respective URLs is needed. These can be generated automatically,
+e.g. with the `webdav4`_ and `fsspec`_ Python libraries.
+
+An example script is given below, using inline comments for explanations.
+
+The example assumes that user's webdav credentials are already known
+to DataLad under the name ``webdav-mycred`` (if not, these can be
+added with ``datalad credentials add``, or provided to the script in a
+different way).
+
+.. _webdav4: https://pypi.org/project/webdav4/
+.. _fsspec: https://pypi.org/project/fsspec/
+
+.. literalinclude:: list_files.py
+   :language: python
+
+This would produce the following csv file:
+
+.. code-block:: none
+
+   name,href
+   file1.dat,/remote.php/dav/files/USERNAME/sharing/example/file1.dat
+   foo/file2.dat,/remote.php/dav/files/USERNAME/sharing/example/foo/file2.dat
+   ...
+	      
+Uncurl remote
+-------------
+
+Download URLs are handled by special remotes. The uncurl remote,
+available in DataLad-next extension, provides both the ability to
+reconfigure URLs and access to DataLad-next's credential workflow. It
+can be initialized as follows (optionally with ``autoenable=true``):
+
```

Comment:
```suggestion
In a DataLad dataset, the process of accessing files that were added via download URLs is handled by a `git-annex special remote`_. The uncurl remote,
available in the `DataLad-next`_ extension, provides both the ability to
reconfigure URLs and access to DataLad-next's credential workflow. It
can be initialized as follows (optionally with ``autoenable=true``) inside a newly created and empty DataLad dataset:

.. _git-annex special remote: https://git-annex.branchable.com/special_remotes/
.. _DataLad-next: https://github.com/datalad/datalad-next

```

Reply:
I understand that you add "inside a newly created and empty DataLad dataset" because the text never suggested that we should be inside a dataset at this stage (good point), but IMO the wording suggests that it must be a newly created and empty dataset, which is not the case (it only has to be before `addurls`, because `addurl` needs to download and checksum the files).

Commiting, but will edit further.